src/Confluent.Kafka/AdminClient.cs

```diff
@@ -657,6 +708,19 @@ public Task DeleteGroupsAsync(IList<string> groups, DeleteGroupsOptions options
             return completionSource.Task;
         }
 
+        /// <summary>
+        ///     Refer to <see cref="Confluent.Kafka.IAdminClient.DeleteConsumerGroupOffsetsAsync(IEnumerable{ConsumerGroupOffsets}, DeleteConsumerGroupOffsetsOptions)" />
+        /// </summary>
+        public Task<List<DeleteConsumerGroupOffsetsResult>> DeleteConsumerGroupOffsetsAsync(IEnumerable<ConsumerGroupOffsets> deleteGroupOffsets, DeleteConsumerGroupOffsetsOptions options = null)
```

Comment:
https://kafka.apache.org/30/javadoc/org/apache/kafka/clients/admin/Admin.html#deleteConsumerGroupOffsets(java.lang.String,java.util.Set)

the java client API only allows working with one group at a time. i think we should do the same - sorry i didn't mention/think of/pick up on this before.

under the hood, librdkafka currently sends separate requests per group, so this is no less efficient with the current implementation. even if we make librdkafka more efficient to send partitions for multiple groups in the one request, this operation is going to be done infrequently, so making the API bigger is the worse tradeoff imo.

.. and we can always add the multi group variant in the future if it comes out as a strong requirement for some reason. that would be slightly annoying as the result and report types would be different, but the naming of that is clear `DeleteConsumerGroupsOffsetsResult` instead of `DeleteConsumerGroupOffsetsResult`.

only make the changes you need - i.e. under the covers support multiple groups, just change the user facing API.

also, it would be good to get to the bottom of the multi-group call not working in your current test before making the changes, in case this is hiding something important.

Reply:
Thanks, will revert the implementation to work for one group at a time.

> also, it would be good to get to the bottom of the multi-group call not working in your current test before making the changes, in case this is hiding something important.

It seems librdkafka is also allowing to modify only a single group at a time([code link](https://github.com/edenhill/librdkafka/blob/75a243f0e41e3566d9398a29e4abfb8139c7e84f/src/rdkafka_admin.c#L3932-L3941)], therefore I think that multi group error is expected.